src/Symfony/Component/Validator/Constraints/ChoiceValidator.php

```diff
@@ -55,6 +55,12 @@ public function validate($value, Constraint $constraint)
                 throw new ConstraintDefinitionException('The Choice constraint expects a valid callback.');
             }
             $choices = $choices();
+            if ($choices instanceof \Traversable) {
```

Comment:
What is the reason for this? What about using `is_iterable()` below (instead of `is_array()`) instead?

Reply:
The reason for this is to handle iterables like collections, yields etc...  in callbacks ([see specific test](https://github.com/symfony/symfony/pull/58615/files#diff-545064a584d44be8aef15f493fc5d8fed17cb1552bffeb19458140d0a6207cb1R42))

That's right I can use `is_iterable()` though.

Reply:
@xabbuh [done](https://github.com/symfony/symfony/pull/58615/commits/e4fba8e42ff91de831bb7c4429abcdbd15953a08#diff-d5a06a06b90acea359eabcd7589da0a955d0852c5b26b7417eb2a95aaa55ec65R59)